crates.io publishing not working

#### Problem

New errors seen when attempting to publish crates for v0.16.5 release.
https://buildkite.com/solana-labs/solana-secondary/builds/3003

There are compilation errors for solana-netutil:
```
Checking solana-netutil v0.16.5
error[E0658]: imports can only refer to extern crate names passed with --extern on stable channel (see issue #53130)
  --> /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/solana-netutil-0.16.5/src/lib.rs:11:9
   |
10 | mod ip_echo_server;
   | ------------------- not an extern crate passed with --extern
11 | pub use ip_echo_server::*;
   |         ^^^^^^^^^^^^^^
   |
note: this import refers to the module defined here
  --> /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/solana-netutil-0.16.5/src/lib.rs:10:1
   |
10 | mod ip_echo_server;
   | ^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try rustc --explain E0658.
error: Could not compile solana-netutil.
```

Also compilation errors for `curve25519-dalek` : 
```
Compiling curve25519-dalek v1.2.1
error[E0658]: Self struct constructors are unstable (see issue #51994)
   --> /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/curve25519-dalek-1.2.1/src/edwards.rs:707:9
    |
707 |         Self(scalar_mul::precomputed_straus::VartimePrecomputedStraus::new(static_points))
    |         ^^^^

error[E0658]: Self struct constructors are unstable (see issue #51994)
   --> /var/lib/buildkite-agent/.cargo/registry/src/github.com-1ecc6299db9ec823/curve25519-dalek-1.2.1/src/ristretto.rs:925:9
    |
925 |         Self(
    |         ^^^^

    Checking env_logger v0.6.2
    Checking rayon v1.1.0
error: aborting due to 2 previous errors

For more information about this error, try rustc --explain E0658.
error: Could not compile curve25519-dalek.
```

--- a/ci/publish-crate.sh
+++ b/ci/publish-crate.sh
@@ -2,6 +2,7 @@
 set -e
 cd "$(dirname "$0")/.."
 source ci/semver_bash/semver.sh
+source ci/rust-version.sh stable
 
 # shellcheck disable=SC2086
 is_crate_version_uploaded() {
@@ -24,8 +25,6 @@ expectedCrateVersion="$MAJOR.$MINOR.$PATCH$SPECIAL"
   echo CRATES_IO_TOKEN undefined
   exit 1
 }
-
-cargoCommand="cargo publish --token $CRATES_IO_TOKEN"
 
 Cargo_tomls=$(ci/order-crates-for-publishing.py)
 
@@ -53,7 +52,7 @@ for Cargo_toml in $Cargo_tomls; do
     crate=$(dirname "$Cargo_toml")
     # TODO: the rocksdb package does not build with the stock rust docker image,
     # so use the solana rust docker image until this is resolved upstream
-    source ci/rust-version.sh
+    cargoCommand="cargo publish --token $CRATES_IO_TOKEN"
     ci/docker-run.sh "$rust_stable_docker_image" bash -exc "cd $crate; $cargoCommand"
   ) || true # <-- Don't fail.  We want to be able to retry the job in cases when a publish fails halfway due to network/cloud issues
 
@@ -67,11 +66,11 @@ for Cargo_toml in $Cargo_tomls; do
       (
         set -x
         rm -rf crate-test
-        cargo init crate-test
+        cargo +"$rust_stable" init crate-test
         cd crate-test/
         echo "${crate_name} = \"${expectedCrateVersion}\"" >> Cargo.toml
         echo "[workspace]" >> Cargo.toml
-        cargo check
+        cargo +"$rust_stable" check
       ) && really_uploaded=1
       if ((really_uploaded)); then
         break;

